dbos/dbos_test.go

```diff
@@ -325,4 +326,111 @@ func TestConfig(t *testing.T) {
 
 		require.NotNil(t, ctx2)
 	})
+
+	t.Run("SystemDBMigrationWithCustomSchema", func(t *testing.T) {
+		t.Setenv("DBOS__APPVERSION", "v1.0.0")
+		t.Setenv("DBOS__APPID", "test-custom-schema")
+		t.Setenv("DBOS__VMID", "test-executor-id")
+
+		customSchema := "dbos_custom_test"
+		ctx, err := NewDBOSContext(context.Background(), Config{
+			DatabaseURL:    databaseURL,
+			AppName:        "test-custom-schema-migration",
+			DatabaseSchema: customSchema,
+		})
+		require.NoError(t, err)
+		defer func() {
+			if ctx != nil {
+				ctx.Shutdown(1 * time.Minute)
+			}
+		}()
+
+		require.NotNil(t, ctx)
+
+		// Get the internal systemDB instance to check tables directly
+		dbosCtx, ok := ctx.(*dbosContext)
+		require.True(t, ok, "expected dbosContext")
+		require.NotNil(t, dbosCtx.systemDB)
+
+		sysDB, ok := dbosCtx.systemDB.(*sysDB)
+		require.True(t, ok, "expected sysDB")
+
+		// Verify schema name was set correctly
+		assert.Equal(t, customSchema, sysDB.schema, "schema name should match custom schema")
+
+		// Verify all expected tables exist in the custom schema
+		dbCtx := context.Background()
+
+		// Test workflow_status table in custom schema
```

Comment:
We should actually try doing something with DBOS and this custom schema instead of just checking a bunch of tables exist

Reply:
Let's add a dedicated custom schema test that actually exercises the system database--particularly send and recv, since the triggers there are the things most likely to break.

Reply:
I'm happy to add a dedicated test that exercises all the system db methods, but, may I ask, why are the triggers special? `pg_notify` works across schema